builders/build/eth-api/dev-env/brownie.md

```diff
@@ -0,0 +1,348 @@
+---
+title: Brownie
+description: 通过Brownie在Moonbeam上使用Python进行编译、部署和调试以太坊式智能合约
+---
+
+# 使用Brownie在Moonbeam上进行部署
+
+![Brownie banner](/images/builders/build/eth-api/dev-env/brownie/brownie-banner.png)
+
+## 概览 {: #introduction }
+
+[Brownie](https://eth-brownie.readthedocs.io/){target=_blank}是一个以太坊开发环境，用于协助Python开发者管理和自动化构建智能合约以及DApp所需的重复性任务。Brownie能够直接与Moonbeam的以太坊API交互，因此其可以用于在Moonbeam上部署智能合约。
+
+本教程将会包含如何在Moonbase Alpha测试网使用Brownie与以太坊式智能合约进行编译、部署和交互。此教程也同样适用于Moonbeam、Moonriver以及Moonbeam开发节点。
+
+## 查看先决条件 {: #checking-prerequisites }
+
+首先，您需要准备以下先决条件：
+
+ - 安装MetaMask并[连接至Moonbase Alpha](/tokens/connect/metamask/){target=_blank}
+ - 拥有一定资金的账户，您可以在[Mission Control](/builders/get-started/moonbase/#get-tokens/){target=_blank}获得所需Token
+ - --8<-- 'text/common/endpoint-examples.md'
+
+## 创建Brownie项目 {: #creating-a-brownie-project }
+
+您将会需要安装Brownie并创建一个Brownie项目（如果您尚未创建）。您可以选择创建一个空白的项目或是使用[Brownie mix](https://eth-brownie.readthedocs.io/en/stable/init.html?highlight=brownie%20mix#creating-a-project-from-a-template){target=_blank}（基础的项目模板）创建项目。本教程将以创建一个空白的项目为例，您可以通过跟随以下步骤进行操作：
+
+1. 为您的项目创建目录
+   
+    ```
+    mkdir brownie && cd brownie
+    ```
+    
+2. 如果您尚未安装`pipx`，执行以下命令进行安装
+   
+    ```
+    python3 -m pip install --user pipx
+    python3 -m pipx ensurepath
+    ```
+    
+2. [使用`pipx`安装Brownie](https://eth-brownie.readthedocs.io/en/stable/install.html){target=_blank}。如果您尚未安装`pipx` ，您可以跟随上个步骤进行安装
+   
+    ```
+    pipx install eth-brownie
+    ```
+    !!! 注意事项
+        [`pipx`](https://github.com/pypa/pipx){target=_blank}用于运行本地安装在您的项目中的可执行文件。Brownie将会被安装在一个虚拟环境中并可在命令行直接使用。
+    
+3. 创建项目
+   
+    ```
+    brownie init
+    ```
+
+![Create Brownie project](/images/builders/build/eth-api/dev-env/brownie/brownie-1.png)
+
+您的Brownie项目应包含以下空白目录：
+
+- **build** —— 用于例如来自编译的合约代码的项目数据
+- **contracts** —— 用于储存智能合约文件
+- **interfaces** —— 用于项目所需的智能合约接口
+- **reports** —— 用于在[Brownie GUI](https://eth-brownie.readthedocs.io/en/stable/gui.html){target=_blank}使用的JSON报告文件
+- **scripts** —— 为Python脚本用于部署合约或是其他自动化事务存在的地方
+- **tests** —— 用于储存测试项目的Python脚本。Brownie将使用`pytest`框架进行单位测试
+
+另外一个不被包含但需要注意且同样重要的文件为`brownie-config.yaml`配置文件。此配置文件为可选项，在自定义特定设置，如默认网络、编译器版本和设置时将会派上用场。
+
+## 网络配置 {: #network-configuration }
+
+要部署至Moonbeam网络，您需要添加并配置网络。
+
+Brownie中的网络配置需通过命令行添加。Brownie可以被用于开发和线上环境。在后台，Brownie使用Ganache作为开发环境。然而，由于Moonbeam开发者节点做为您专属的开发环境，您并不需要Ganache。因此，您可以使用线上网络（如Moonriver和Moonbase Alpha）所使用的配置来配置您的节点。
```

Comment:
this section seems like it might be missing something, can you please double check it?

Reply:
Looks like it's missing these two lines: 

```
Moonbeam is supported out of the box with Brownie, but if you're looking to deploy a contract to Moonriver, Moonbase Alpha, or a Moonbeam development node you'll need to add the network configurations.

If you're familiar with Brownie, you're probably used to the `Mainnet` and `Testnet` network setup. To be consistent, you can add Moonbase Alpha and Moonbeam development node configurations nestled under the pre-existing Moonbeam network configuration. Since Moonriver is it's own MainNet, you can add an entirely new configuration for Moonriver.
```